photutils/psf/psfphotometry.py

```diff
@@ -0,0 +1,446 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Module which provides classes to perform PSF Photometry"""
+
+from __future__ import division
+import abc
+import numpy as np
+from astropy.extern import six
+from astropy.table import Table, vstack
+from astropy.modeling.fitting import LevMarLSQFitter
+from astropy.nddata.utils import overlap_slices, NoOverlapError
+from astropy.stats import gaussian_sigma_to_fwhm
+from ..psf import subtract_psf
+from ..aperture import CircularAperture, aperture_photometry
+
+
+__all__ = ['PSFPhotometryBase', 'DAOPhotPSFPhotometry']
+
+
+@six.add_metaclass(abc.ABCMeta)
+class PSFPhotometryBase(object):
```

Comment:
For now I think you should do away with the base class here.  I think it's not clear that the `do_photometry` function would look the same for other approaches.  Unlike things like the star-finders and groupers, the "main task" here is a bit less clear.  And because this is the "highest level" task, it's less likely to need to have a fromalized interface.